### ISSUE TITLE
Load LLVM for clang containers.

### ISSUE THREAD

**Issue by mdavis36** [rse-ops/docker-images#85]
When using clang compilers you need to load LLVM before running codes built with omp support. It would be nice if LLVM pre-loaded in the containers.

**Comment by vsoch** [comment]
We could definitely try that! @trws @davidbeckingsale what would that look like w.r.t spack?

**Comment by davidbeckingsale** [comment]
I think we could do it in the base images making sure LD_LIBRARY_PATH is set up correctly to pick up what's added to the view, extending what's done here: https://github.com/rse-ops/docker-images/blob/main/ubuntu/22.04/Dockerfile#L11

**Comment by trws** [comment]
Yeah, I had intended that to work based on those changes, must be something missing.

**Comment by trws** [comment]
Rather these changes: https://github.com/rse-ops/docker-images/blob/1e2bb6a1322ecbaa0d5497ef0e9c13794e8003e7/scripts/set-up-spack.sh#L37

What directory is missing from the library search? We should add something to the clang file that adds it to the load cache, that'll be more robust than the environment variable or loading would be.

**Comment by trws** [comment]
Found the bug, we were only getting lib64 in the search path instead of both lib and lib64, fix is coming shortly.

**Comment by mdavis36** [comment]
@vsoch @trws This issue has popped up for us again, we are having to manually add /opt/view/lib to LD_LIBRARY_PATH for our clang image builds.

**Comment by trws** [comment]
In what image/context? The path should always be there now, it's still added to ld.so.conf and reconfigured, so something funny must be happening.

**Comment by vsoch** [comment]
Could you link the image? it could be an old one - I was perusing this weekend and we still have some old images in the UI (and perhaps we should remove).

**Comment by mdavis36** [comment]
We were seeing this using the below images as our base image:

ghcr.io/rse-ops/clang-ubuntu-20.04:llvm-11.0.0
ghcr.io/rse-ops/clang-ubuntu-22.04:llvm-13.0.0

I fixed it in this PR https://github.com/LLNL/RAJA/pull/1301

**Comment by vsoch** [comment]
@trws it looks like we do ldconfig in our base image, should we try adding it in the images referenced as a final command?

https://github.com/rse-ops/docker-images/blob/cd590b3e95b0489eef14fe14069f601497607323/scripts/set-up-spack.sh

**Comment by trws** [comment]
We could, but if ldconfig has run, it shouldn't need to be run again unless the new file is removed and ldconfig is run again.  Not sure what would be causing a problem with it.

**Comment by vsoch** [comment]
Could we just add to LD_LIBRARY_PATH too then?

**Comment by trws** [comment]
I don’t see any harm in that, should just be redundant but if it helps then great!


---
Sent from Workspace ONE Boxer<https://whatisworkspaceone.com/boxer>

On August 16, 2022 at 8:55:05 PM PDT, Vanessasaurus ***@***.***> wrote:

Could we just add to LD_LIBRARY_PATH too then?
